### PR TITLE
 Small sky refactor to fix stereo issues

### PR DESCRIPTION
**Pull request by fabien-unity**
### Purpose of this PR
Avoid duplicate code and fix issue with stereo sky rendering

---
### Testing status
**Katana Tests**: [green on Windows](https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?automation-tools_branch=add-platform-filter&ScriptableRenderLoop_branch=xr%2Fhdrp%2Fsky&unity_branch=trunk)

**Manual Tests**: Windows 10 with Oculus Rift and 1080Ti (test 1502 and 2007)

---
### Overall Product Risks
**Technical Risk**: None

**Halo Effect**: None

---
### Comments to reviewers
I've added a new file SkyUtils.h to avoid duplicate code.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/Sky/ProceduralSky/ProceduralSky.shader [hunk 1]
```diff
@@ -120,22 +121,15 @@ Shader "Hidden/HDRP/Sky/ProceduralSky"
...
+
         // Reverse it to point into the scene
         float3 dir = -viewDirWS;
-#endif
 
         // Rotate direction
-        float phi = DegToRad(_SkyParam.z);
-        float cosPhi, sinPhi;
-        sincos(phi, sinPhi, cosPhi);
+        // TODO: remove this code? the rotation value from the CPU is always 0
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Sky/ProceduralSky/ProceduralSky.shader]: yes, remove it, useless :)

**Comment by sebastienlagarde** [comment]
Just remove the useless code and we are good to go :)

**Comment by sebastienlagarde** [comment]
Note: you have various graphic failing and this isn't expected (Master is green). Try to run them locally (open HDRP_Test, launch test runner). To see what is happening

File: com.unity.render-pipelines.high-definition/Runtime/Sky/ProceduralSky/ProceduralSky.shader (hunk 1 above)
Comment (fabien-unity) [reply com.unity.render-pipelines.high-definition/Runtime/Sky/ProceduralSky/ProceduralSky.shader]: Alright, done.

**Comment by fabien-unity** [comment]
I'll look into the failing tests.

**Comment by fabien-unity** [comment]
Tests are now green on Windows. Mac is unavailable for testing due to lack of machines.

**Pull request merged**